### PR TITLE
Fix backspace crash on Chinese devices

### PR DESCRIPTION
**Pull request by GaryQian**
The Selection.extendLeft and extendRight calls on Huawei devices has a bug where it will ignore the selection indexes and always use the initial length of the editable. This works around it by falling back to a simpler algorithm.

### CODE REVIEW

File: shell/platform/android/io/flutter/plugin/editing/InputConnectionAdaptor.java [hunk 1]
```diff
@@ -168,10 +168,20 @@ public boolean sendKeyEvent(KeyEvent event) {
...
-                    } else {
-                        Selection.extendLeft(mEditable, mLayout);
+                    try {
+                        if (isRtl) {
+                            Selection.extendRight(mEditable, mLayout);
+                        } else {
+                            Selection.extendLeft(mEditable, mLayout);
+                        }
+                    } catch (IndexOutOfBoundsException e) {
+                        // On Huaewi devices on initial app startup before focus is lost,
```
Comment (jason-simmons) [inline shell/platform/android/io/flutter/plugin/editing/InputConnectionAdaptor.java]: typo: "Huawei"

**Comment by GaryQian** [comment]
Partially fixes https://github.com/flutter/flutter/issues/33642

**Comment by CaiJingLong** [comment]
It also happens on the xiaomi device, probably due to the security keyboard.

1. Enter information in the username input box.
2. Switch focus to the password input box, the system's security keyboard pops up. Enter the password.
3. Switch focus to the username input box, delete 1~2 characters, crash.

My record screen video : https://drive.google.com/file/d/1jJRTjHvXu1bmSaannWfEtlGD8Q-VTMoJ/view?usp=sharing

**Comment by GaryQian** [comment]
Either way, this should fix the crashing. It turns out it may not be unique to only Huawei, as there were a few reports on xiaomi devices.

**Comment by GaryQian** [comment]
The root bug is actually in the affected devices. This is a workaround. Until the root bug is fixed, I'm not sure if we can easily reach 100% correct backspace behavior here. Once the app loses focus for the first time, the problem is fixed.

When calling extendLeft and extendRight, it seems that the implementation under some conditions fails to update some internal index or buffer, causing it to perform all extend (and move) operations based off of the initial condition of the text, completely ignoring any selection changes. When the text becomes shorter than the initial contents, the crash comes from trying to access text that no longer exists.

Since the Android API is not doing what it promises, and the bug prevents working with selections to fix it, I dont think a very straightforwards full solution exists. Attempts at handling this explicitly result in also breaking emoji backspace support.

**Pull request merged**